packages/react-devtools-shared/src/__tests__/console-test.js

```diff
@@ -524,12 +524,44 @@ describe('console', () => {
       return <div />;
     }
 
-    patchConsole({
-      appendComponentStack: false,
-      breakOnWarn: false,
-      showInlineWarningsAndErrors: false,
-      hideConsoleLogsInStrictMode: true,
-    });
+    act(() =>
+      root.render(
+        <React.StrictMode>
+          <App />
+        </React.StrictMode>,
+      ),
+    );
+
+    expect(mockLog).toHaveBeenCalledTimes(1);
+    expect(mockLog.mock.calls[0]).toHaveLength(1);
+    expect(mockLog.mock.calls[0][0]).toBe('log');
+
+    expect(mockWarn).toHaveBeenCalledTimes(1);
+    expect(mockWarn.mock.calls[0]).toHaveLength(1);
+    expect(mockWarn.mock.calls[0][0]).toBe('warn');
+
+    expect(mockError).toHaveBeenCalledTimes(1);
+    expect(mockError.mock.calls[0]).toHaveLength(1);
+    expect(mockError.mock.calls[0][0]).toBe('error');
+  });
+
+  it('should double log in Strict mode initial render for extension', () => {
+    global.__REACT_DEVTOOLS_APPEND_COMPONENT_STACK__ = false;
+    global.__REACT_DEVTOOLS_HIDE_CONSOLE_LOGS_IN_STRICT_MODE__ = false;
+
+    global.__REACT_DEVTOOLS_GLOBAL_HOOK__.rendererInterfaces.set(
```

Comment:
I think you're doing this to simulate a render that's happening before the DevTools and React DOM have done their handshake?

(Otherwise, this looks like the same test as "_should not double log if hideConsoleLogsInStrictMode is enabled in Strict mode_".)

This could use an inline comment 😁